gobblin-config-management/gobblin-config-client/src/test/java/gobblin/config/client/TestConfigClient.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright (C) 2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.config.client;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.mockito.Mockito;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+import com.typesafe.config.ConfigValue;
+
+import gobblin.config.client.api.VersionStabilityPolicy;
+import gobblin.config.common.impl.ConfigStoreValueInspector;
+import gobblin.config.common.impl.SingleLinkedListConfigKeyPath;
+import gobblin.config.store.api.ConfigKeyPath;
+import gobblin.config.store.api.ConfigStore;
+import gobblin.config.store.api.ConfigStoreFactory;
+
+@Test(groups = { "gobblin.config.common.impl" })
+
+public class TestConfigClient {
```

Comment:
It is very hard to figure out what's going on here. For example, it is hard to figure out what are the config keys and values in the mock store. Based on looking at the implementation, I am not sure the handling of relative config keys is correct but I can't figure out how the test works.

I think it would have been useful to have a simple store implementation which is just a map from a config key to a Properties/Config.